flink-yarn-tests/src/test/java/org/apache/flink/yarn/YarnTestBase.java

```diff
@@ -212,30 +212,45 @@ public void checkClusterEmpty() {
 	 * Sleep a bit between the tests (we are re-using the YARN cluster for the tests).
 	 */
 	@After
-	public void sleep() throws IOException, YarnException {
-		Deadline deadline = Deadline.now().plus(Duration.ofSeconds(10));
+	public void shutdownYarnClient() {
+		yarnClient.stop();
+	}
 
-		boolean isAnyJobRunning = yarnClient.getApplications().stream()
-			.anyMatch(YarnTestBase::isApplicationRunning);
+	protected void runTest(RunnableWithException test) {
+		Throwable testFailure = null;
+		try {
+			test.run();
+		} catch (Throwable t) {
+			testFailure = t;
+		}
 
-		while (deadline.hasTimeLeft() && isAnyJobRunning) {
-			try {
-				Thread.sleep(500);
-			} catch (InterruptedException e) {
-				Assert.fail("Should not happen");
-			}
-			isAnyJobRunning = yarnClient.getApplications().stream()
+		try {
```

Comment:
I think this block should be the `finally` block of the `try { test.run() } finally { ... }`

Reply:
What value would that add? We'd still need both try-catch blocks.

Reply:
Why do we need both try-catch blocks? I think we don't need this.